src/MultipartBody.cs

```diff
@@ -148,7 +148,14 @@ public void Serialize(ISerializationWriter writer)
                 }
                 else if(part.Content is string currentString)
                 {
-                    writer.WriteStringValue(string.Empty, currentString);
+                    using var partWriter = RequestAdapter.SerializationWriterFactory.GetSerializationWriter(part.ContentType);
+                    partWriter.WriteStringValue(string.Empty, currentString);
+                    using var partContent = partWriter.GetSerializedContent();
```

Comment:
can we move lines 153 to 158 to a private function so that we can also elimintate the repetition in the if block above?